plugins/events-backend-module-sqs/src/service/AwsSqsEventPublisher.ts

```diff
@@ -0,0 +1,149 @@
+ *
+ * @public
+ */
+export class AwsSqsEventPublisher implements EventPublisher {
+  private readonly topic: string;
+  private readonly receiveParams: ReceiveMessageCommandInput;
+  private readonly sqs: SQSClient;
+  private readonly queueUrl: string;
+  private eventBus?: EventBus;
+
+  static fromConfig(env: {
+    config: Config;
+    logger: Logger;
+    scheduler: PluginTaskScheduler;
+  }): AwsSqsEventPublisher[] {
+    return readConfig(env.config).map(
+      config => new AwsSqsEventPublisher(env.logger, env.scheduler, config),
+    );
+  }
+
+  private constructor(
+    private readonly logger: Logger,
+    private readonly scheduler: PluginTaskScheduler,
+    config: AwsSqsEventSourceConfig,
+  ) {
+    this.topic = config.topic;
+    this.receiveParams = {
+      MaxNumberOfMessages: 10,
+      QueueUrl: config.queueUrl,
+      VisibilityTimeout: config.visibilityTimeout,
+      WaitTimeSeconds: config.waitTime,
+    };
+    this.sqs = new SQSClient({ region: config.region });
+    this.queueUrl = config.queueUrl;
+  }
+
+  async connect(eventBus: EventBus): Promise<void> {
+    this.eventBus = eventBus;
+    return this.scheduleTask();
+  }
+
+  private async scheduleTask(): Promise<void> {
+    const id = `events.awsSqs:${this.topic}`;
+    const logger = this.logger.child({
+      class: AwsSqsEventPublisher.prototype.constructor.name,
+      taskId: id,
+    });
+
+    await this.scheduler.scheduleTask({
+      id: id,
+      frequency: { minutes: 1 },
+      timeout: { minutes: 3 },
+      fn: async () => {
+        try {
+          await this.run();
+        } catch (error) {
+          logger.error(error);
+        }
+      },
+    });
+  }
+
+  private async deleteMessages(messages?: Message[]): Promise<void> {
+    if (!messages) {
+      return;
+    }
+
+    const deleteParams = {
+      QueueUrl: this.queueUrl,
+      Entries: messages.map((message, index) => {
+        return {
+          Id: message.MessageId ?? `message-${index}`,
+          ReceiptHandle: message.ReceiptHandle,
+        };
+      }),
+    };
+
+    try {
+      const result = await this.sqs.send(
+        new DeleteMessageBatchCommand(deleteParams),
+      );
+      if (result.Failed) {
+        this.logger.error(
+          `Failed to delete ${result.Failed!.length} of ${
+            messages.length
+          } messages from AWS SQS ${this.queueUrl}. First: ${
+            result.Failed[0].Message
+          }`,
+        );
+      }
+    } catch (error) {
+      this.logger.error(
+        `Failed to delete message from AWS SQS ${this.queueUrl}`,
+        error,
+      );
+    }
+  }
+
+  private async run() {
+    try {
+      const data = await this.sqs.send(
+        new ReceiveMessageCommand(this.receiveParams),
+      );
+
+      data.Messages?.forEach(message => {
+        const event = JSON.parse(message.Body!);
+        this.eventBus!.publish(this.topic, event);
```

Comment:
This would make the whole scheduling part unnecessary then and every instance with it installed would start consuming. Probably even better as the background tasks wait for task to complete for another one can be started which would not really scale. 🤔